### PR TITLE
fix: Guild.query_members may accept empty query and limit

### PR DESCRIPTION
**Pull request by NeloBlivion**
## Summary

When requesting guild members [through the gateway](https://discord.com/developers/docs/topics/gateway-events#request-guild-members), you can pass an empty string `query` and `limit=0` to recieve all members. This adjusts `Guild.query_members` to accept `limit=None` (or `0`) to allow this behavior, though the default `limit` isn't changed to avoid breaking, and interprests `query=None` as an empty string.
As such, all guild members can be quickly recieved (and cached) through `await guild.query_members(limit=None)`.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by plun1331 — CHANGES_REQUESTED** [review]
please :clap: add 👏 a 👏 change :clap: log :clap: entry

File: discord/guild.py [hunk 1]
```diff
@@ -3384,20 +3384,25 @@ async def query_members(
...
+                    "limit must be between 1 and 100 when using query or user_ids"
+                )
+            if not limit:
+                query = ""
+                limit = 0
 
+        if query is None:
             if user_ids is None:
-                raise ValueError("Must pass either query or user_ids")
+                raise ValueError("Must pass query, user_ids, or set limit to None")
```
Comment (JustaSqu1d) [inline discord/guild.py:3398]:
```suggestion
                raise ValueError("Must pass query or user_ids, or set limit to None")
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

**Review by JustaSqu1d — APPROVED** [review]

**Review by Lulalaby — DISMISSED** [review]

File: discord/guild.py [hunk 2]
```diff
@@ -3379,22 +3379,22 @@ async def query_members(
...
-            The maximum number of members to send back. This must be
-            a number between 5 and 100.
-        presences: :class:`bool`
+        user_ids: Optional[List[:class:`int`]]
+            List of user IDs to search for. If the user ID is not in the guild then it won't be returned.
+
+            .. versionadded:: 1.4
+        limit: Optional[:class:`int`]
+            The maximum number of members to send back. If no query is passed, pass ``None`` to return all members.
+            If you specify a ``query`` or ``user_ids``, this must be between 1 and 100. Defaults to 5.
```
Comment (Dorukyum) [inline discord/guild.py:3388]:
```suggestion
            The maximum number of members to send back. If no query is passed, passing ``None`` returns all members.
            If a ``query`` or ``user_ids`` is passed, must be between 1 and 100. Defaults to 5.
```

File: discord/guild.py [hunk 3]
```diff
@@ -3379,22 +3379,22 @@ async def query_members(
...
+        presences: Optional[:class:`bool`]
             Whether to request for presences to be provided. This defaults
             to ``False``.
 
             .. versionadded:: 1.6
 
-        cache: :class:`bool`
+        cache: Optional[:class:`bool`]
```
Comment (Dorukyum) [inline discord/guild.py:3395]: This is not optional. It defaults to `True` which is a bool.
```suggestion
        cache: :class:`bool`
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by plun1331 — APPROVED** [review]

**Review by JustaSqu1d — APPROVED** [review]

**Pull request merged**